clang_tools/util.py

```diff
@@ -63,3 +64,35 @@ def download_file(url: str, file_name: str) -> Optional[str]:
     file = Path(file_name)
     file.write_bytes(buffer)
     return file.as_posix()
+
+
+def get_sha_checksum(binary_url: str) -> str:
+    """Fetch the SHA512 checksum corresponding to the released binary.
+
+    :param binary_url: The URL used to download the binary.
+
+    :returns: A `str` containing the contents of the SHA512sum file given
+        ``binary_url``.
+    """
+    with urllib.request.urlopen(
+        binary_url.replace(".exe", "") + ".sha512sum"
+    ) as response:
+        response: HTTPResponse
+        return response.read(response.length).decode(encoding="utf-8")
+
+
+def verify_sha512(checksum: str, exe: bytes) -> bool:
+    """Verify the executable binary's SHA512 hash matches the valid checksum.
+
+    :param checksum: The SHA512 checksum.
+    :param exe: The `bytes` content of the binary executable that is to be verified.
+
+    :returns: `True` if the ``exe`` hash matches the ``checksum`` given,
+        otherwise `False`.
+    """
+    if " " in checksum:
+        # released checksum's include the corresponding filename (which we don't need)
+        checksum = checksum.split(" ", 1)[0]
+    valid_hash = int(checksum, 16).to_bytes(int(len(checksum) / 2), "big")
+    bin_hash = hashlib.sha512(exe).digest()
```

Comment:
If using `bin_checksum = hashlib.sha512(exe).hexdigest()` instead of `bin_hash = hashlib.sha512(exe).digest()`, don't need to convert checksum to bytes, compared `checksum == bin_checksum` should more readble.

Reply:
A str object with hexadecimal characters is not the same as a bytes object. My dev tests require the string be interpreted as a hexadecimal number then converted to a bytes object (using big endianness).

```py
>>> s = "DEADBEEF"
>>> h = 0xDEADBEEF
>>> s == h
False
>>> b = b"\xDE\AD\xBE\xEF"
>>> s == b
False
>>> h == b
False
````

Reply:
oh I didn't see that [`hexdigest()` method](https://docs.python.org/3/library/hashlib.html#hashlib.shake.hexdigest). I think you're right. Let me try it...

Reply:
Good suggestion. Your request has been granted.